ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -129,7 +143,19 @@ kern_return_t KauthHandler_Init()
     {
         goto CleanupAndFail;
     }
-    
+
+
+    s_renameLock = SpinLock_Alloc();
+    s_maxPendingRenames = 8; // Arbitrary choice, should be maximum number of expected concurrent threads performing renames, but array will resize on demand
+    s_pendingRenames = Memory_AllocArray<PendingRenameOperation>(s_maxPendingRenames);
+    if (!SpinLock_IsValid(s_renameLock) || s_pendingRenames == nullptr)
+    {
+        goto CleanupAndFail;
+    }
+
+    Array_DefaultInit(s_pendingRenames, s_maxPendingRenames);
```

Comment:
Curious why we fill the array with objects rather than allowing empty slots to be null.

Reply:
In C++, initialising a variable of type `T` with `T()` or `T{}` means it is "value-initialised", which ends up being zero-initialisation for primitive (member) types (otherwise, primitive members are zero-initialised, then the default constructor is called). So in this case `PendingRenameOperation()` means constructing a `PendingRenameOperation` which has both its fields (which are pointers, so primitive types) zero-initialised. (i.e. `.vnode` and `.thread` are both `nullptr`, which is what we want.)